### PR TITLE
feat(tests): Add individual evergreen test results with XUnit

### PR DESCRIPTION
**Pull request by gagik**
Allows for individual testing results to be viewed and makes it easier to detect specific flaky tests with Foliage in the future.

### CODE REVIEW

File: .evergreen.yml [hunk 1]
```diff
@@ -12448,6 +12516,7 @@ tasks:
           executable_os_id: win32
       - func: run_e2e_tests
         vars:
+          task_name_var: "${task_name}"
```
Comment (addaleax) [inline .evergreen.yml:12519]: Keep in mind that this file is automatically generated from `.evergreen/evergreen.yml.in` 🙂

Comment (gagik) [reply .evergreen.yml:12519]: Oh! so are my changes not doing anything at all? I have suspicion that `${task_name}` is always empty right now for the `run_e2e_tests` and have been battling it for so long

Comment (gagik) [reply .evergreen.yml:12519]:
```
npm run update-evergreen-config
```
oh cool!

Comment (addaleax) [reply .evergreen.yml]: Yeah, although any changes you make to `.evergreen.yml` should already have an effect (they'd just get overwritten after merging).

I don't know what exactly's going on, but you shouldn't need the `task_name_var` indirection here, using `${task_name}` directly in the `env:` list should already work

Comment (gagik) [reply .evergreen.yml]: Yeah, I am suspecting this expansion is either not working or somehow getting lost in the process. I looked through evergreen docs and nothing suspicious there.

Comment (gagik) [reply .evergreen.yml]: Okay seems like the env variable just somehow doesn't reach the bash script

**Comment by gagik** [comment]
evergreen validate -p mongosh

File: .logs/empty.xml [hunk 2]
```diff
@@ -0,0 +1,3 @@
+<!-- An empty test suite which is needed as not all tasks produce XUnit results and the current Evergreen setup always expects some file to be uploaded -->
+<testsuite name="empty" tests="0" failures="0" errors="0" skipped="0" timestamp="Fri, 18 Oct 2024 09:21:06 GMT" time="8.907">
```
Comment (gagik) [inline .logs/empty.xml:2]: I can't think of a better way to go about this... evergreen seems to fail the job completely when no files are found in the attach results step

**Review by addaleax — APPROVED** [review]

**Pull request merged**